model/sharing/upload.go

```diff
@@ -350,6 +366,43 @@ func (s *Sharing) uploadFile(inst *instance.Instance, m *Member, file map[string
 	return nil
 }
 
+func onSameStack(src, dst *instance.Instance) bool {
+	var srcPort, dstPort string
+	parts := strings.SplitN(src.Domain, ":", 2)
+	if len(parts) > 1 {
+		srcPort = parts[1]
+	}
+	parts = strings.SplitN(dst.Domain, ":", 2)
+	if len(parts) > 1 {
+		dstPort = parts[1]
+	}
+	return srcPort == dstPort
+}
```

Comment:
Cannot this be replaced with `src.InstanceHost() == dst.InstanceHost()`?

Reply:
You're right. I read that function too quickly (I read "host" as machine host rather than instance host).